py/server/deephaven/table.py

```diff
@@ -407,19 +410,61 @@ def _sort_column(col, dir_):
         _JColumnName.of(col)))
 
 
-def _td_to_columns(table_definition):
-    cols = []
-    j_cols = table_definition.getColumnsArray()
-    for j_col in j_cols:
-        cols.append(
-            Column(
-                name=j_col.getName(),
-                data_type=dtypes.from_jtype(j_col.getDataType()),
-                component_type=dtypes.from_jtype(j_col.getComponentType()),
-                column_type=ColumnType(j_col.getColumnType()),
-            )
-        )
-    return cols
+class TableDefinition(JObjectWrapper,Mapping):
+    """A Deephaven table definition."""
+    j_object_type = _JTableDefinition
+
+    def __init__(self, j_table_definition: jpy.JType):
+        self.j_table_definition = j_table_definition
+
+    @property
+    def j_object(self) -> jpy.JType:
+        return self.j_table_definition
+
+    @property
```

Comment:
`@cached_property`?

Reply:
It's not clear to me if we want to cache _Table_; this would mean that the table would exist until TableDefinition is GCd.

Reply:
I concur with Devin.